src/xtgeoviz/plot/xsection.py

```diff
@@ -379,13 +379,13 @@ def plot_well(
         wo = self._well
 
         # reduce the well data by Pandas operations
-        dfr = wo.dataframe
-        wo.dataframe = dfr[dfr["Z_TVDSS"] > self._zmin]
+        dfr = wo.get_dataframe()
+        wo.set_dataframe(dfr[dfr["Z_TVDSS"] > self._zmin])
 
         # Create a relative XYLENGTH vector (0.0 where well starts)
         wo.create_relative_hlen()
 
-        dfr = wo.dataframe
+        dfr = wo.get_dataframe()
         if dfr.empty:
             self._showok = False
             return
```

Comment:
Here we could simplify even more:

`if wo.get_dataframe(copy=False).empty:  ...`

Reply:
It seems we actually need to refresh the local variable again after running `create_relative_hlen()`